common_code/inspiral/CE_orbit.f90

```diff
@@ -173,14 +294,32 @@ subroutine check_merger(id, ierr)
          CE_companion_radius = s% xtra(3)
          CE_companion_mass = s% xtra(4)
 
-         rl_core = eval_rlobe(s% he_core_mass,CE_companion_mass,CE_companion_position)*CE_companion_position
-         rl_companion = eval_rlobe(CE_companion_mass,s% he_core_mass,CE_companion_position)*CE_companion_position
+         CE_companion_molec_weight = s% x_ctrl(19)
+
+         rl_core = eval_rlobe(s% he_core_mass,CE_companion_mass,CE_companion_position)
+         rl_companion = eval_rlobe(CE_companion_mass,s% he_core_mass,CE_companion_position)
+
+          ! Check if the star companion can be disrupted by the enclosed mass
+         rl_companion_enc = eval_rlobe(CE_companion_mass,s% xtra(15) / Msun,CE_companion_position)
+
+         ! Compute companion's virial temperature 
+         virial_temp = CE_companion_molec_weight * mp * standard_cgrav * CE_companion_mass * Msun / ( boltzm * CE_companion_radius * Rsun )
+
 
          ! Merger condition
          ! Merge if either the companion or the core fills its rochelobe
          if ((rl_companion .le. CE_companion_radius) .or. (rl_core .le. s% he_core_radius)) then
             s% lxtra(1) = .true.
             s% xtra(2) = 0.0
+            write(*,*) "!!!!!!!!!!!! Either the core or the companion filled their Roche Lobe !!!!!!!!!!!!"
+         else if (rl_companion_enc .le. CE_companion_radius) then 
+            s% lxtra(1) = .true.
+            s% xtra(2) = s% xtra_old(2)
+            write(*,*) "!!!!!!!!!!!! Companion filled its Roche Lobe !!!!!!!!!!!!"
```

Comment:
Is there any possible case, where `rl_companion<rl_companion_enc`? Hence, do we need both or shouldn't the new one replace the old one?